util/member.js

```diff
@@ -0,0 +1,63 @@
+import { pinJSONToIPFS } from './pinata'
+import { getIpfsData } from './ipfs'
+
+const simpleMergeMethod = (existingData, newData) => ({
+  ...existingData,
+  ...newData,
+})
+
+const updateMemberData = async (
+  integrationName,
+  newData,
+  mergeMethod,
+  currentIpfsHash = null,
+) => {
+  // first get any past member data
+  let memberData = {}
+  if (currentIpfsHash) {
+    memberData = await getIpfsData(currentIpfsHash)
+  }
+
+  const existingData = memberData[integrationName] || {}
+
+  // merge in new data
+  memberData = {
+    ...memberData,
+    [integrationName]: mergeMethod(existingData, newData),
+  }
+
+  // and then resave the data to IPFS
+  const { IpfsHash } = await pinJSONToIPFS(memberData)
+  return IpfsHash
+}
+
+export const mergeMembersData = async (
+  integrationName,
+  membersDataHashes,
+  data,
+  mergeMethod = simpleMergeMethod,
+) => {
+  // Get list of member addresses that need to be updated
+  const incomingDataAddresses = Object.keys(data)
+
+  // Update each member’s data
+  const updatedIpfsHashes = await Promise.all(
+    incomingDataAddresses.map(async address =>
+      updateMemberData(
+        integrationName,
+        data[address],
+        mergeMethod,
+        membersDataHashes[address],
+      ),
+    ),
+  )
+
+  // Return updated members data hashes
+  return incomingDataAddresses.reduce(
+    (updatedMembersDataHashes, address, index) => ({
+      ...updatedMembersDataHashes,
+      [address]: updatedIpfsHashes[index],
+    }),
+    membersDataHashes,
+  )
+}
```

Comment:
Thinking it might be nice to nest API specific util files under `utill/api/*` thoughts?

The other day someone accidentally imported a util meant for the API in the client side and it broke because it was expecting private runtime values.

Reply:
Good point! Will move.